tests/mocks/mock_tool/tool.py

```diff
@@ -52,6 +53,15 @@ def test_str_output(self, value: dict) -> str:
     def test_no_schema(self, value: dict) -> str:
         return "no schema"
 
+    @activity(
+        config={
+            "description": "test description",
+            "schema": lambda _self: Schema(_self.custom_schema, description="Test input"),
+        }
+    )
```

Comment:
This is awesome, but why not take it further and just pass a lambda into the decorator directly?

Reply:
That's what I originally wanted to do but it's not straightforward (if possible at all) to provide a `_self` at that stage.